src/main/java/ratpack/zipkin/internal/ZipkinHttpClientImpl.java

```diff
@@ -105,14 +109,63 @@ public void close() {
         return request(uri, requestConfigurer.prepend(RequestSpec::post));
     }
 
-    private Action<? super RequestSpec> tracedRequestAction(final Action<? super RequestSpec> action) {
-        return action.append(requestSpec -> {
-            MethodCapturingRequestSpec captor = new MethodCapturingRequestSpec(requestSpec);
+    private Action<? super RequestSpec> actionWithSpan(final Action<? super RequestSpec> action, final Span span) {
+        return (request -> {
+            MethodCapturingRequestSpec captor = new MethodCapturingRequestSpec(request);
+            span.start();
             action.execute(captor);
-            HttpMethod capturedMethod = Optional.ofNullable(captor.getCapturedMethod()).orElse(HttpMethod.GET);
-            requestInterceptor
-                    .handle(requestAdapterFactory.createAdaptor(requestSpec.method(capturedMethod), capturedMethod.getName()));
+            span.name(spanNameProvider.getName(new DefaultRequestSpecNameAdapter(captor)));
+            span.tag(TraceKeys.HTTP_URL, captor.getUri().toString());
         });
     }
 
+    private void streamedResponseWithSpan(Result<StreamedResponse> response, Span span) {
+        resultWithSpan(response, () -> response.getValue().getStatusCode(), span);
+    }
+
+    private void responseWithSpan(Result<ReceivedResponse> response, Span span) {
+        resultWithSpan(response, () -> response.getValue().getStatusCode(), span);
+    }
+
+    private void resultWithSpan(Result<?> result, StatusCode statusCode, Span span) {
+        if (result.isError()) {
+            String message = result.getThrowable().getMessage();
+            if (message != null) {
+                span.tag(Constants.ERROR, result.getThrowable().getClass().getSimpleName());
+            }
+        } else {
+            int status = statusCode.getStatusCode();
```

Comment:
cs/cr annotations are handled by the tracer.  Thanks for bringing it up though because I did forget to set the span kind to client here.